synapse/storage/schema/delta/53/user_share.sql

```diff
@@ -0,0 +1,49 @@
+/* Copyright 2017 Vector Creations Ltd, 2019 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+DROP TABLE users_who_share_rooms;
+
+-- This is no longer used because it's duplicated by the users_who_share_public_rooms
+DROP TABLE users_in_public_rooms;
+
+-- Table keeping track of who shares a room with who. We only keep track
+-- of this for local users, so `user_id` is local users only (but we do keep track
+-- of which remote users share a room)
```

Comment:
can you try and clarify what exactly is stored in this table? For example, if Alice and Bob are both local users, will there be two entries for each room they share (one with `user_id == alice` and one with `user_id == bob`) ? What if one of them is a remote user?

Reply:
or will there even be four entries? (`alice, alice`, `alice, bob`, `bob, alice`, `bob, bob`)

Reply:
There will be four entries. Hopefully the new comments spell it out a bit better.